core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -282,6 +299,7 @@ protected TableMetadata refresh() {
   @Override
   public void commit() {
     // this is always set to the latest commit attempt's snapshot id.
+    String branch = toBranch == null ? SnapshotRef.MAIN_BRANCH : toBranch;
```

Comment:
If we set the default value of targetBranch as main, then the branch to produce snapshots to just becomes targetBranch. Then we don't need to declare a separate variable, and so we can remove this line

Reply:
Agree with your suggestion. I have defaulted targetBranch to main and just changed the way we access snapshot by ref.